packages/input/src/index.vue

```diff
@@ -1,20 +1,37 @@
 <template>
 	<div class="t-ui-input">
-		<label :for="id" class="t-ui-input__label">
+		<label :for="id" class="t-ui-input__label" data-test="label">
 			{{ label }}
 			<slot name="label-text" />
 		</label>
-		<input
+		<p
+			v-if="description && descriptionPosition === 'top'"
+			:id="`${id}__desc`"
+			class="t-ui-input__description"
+			data-test="description-top"
+		>
+			{{ description }}
+		</p>
+		<component
+			:is="isTextarea ? 'textarea' : 'input'"
 			:id="id"
 			:value="value"
 			:aria-invalid="validation.$error.toString()"
 			:aria-describedby="ariaDescribedby"
 			v-bind="$attrs"
 			class="t-ui-input__input"
+			data-test="input"
 			v-on="listeners"
 			@input="updateValue"
-		/>
-		<p v-if="description" :id="`${id}__desc`" class="t-ui-input__description">
+		>
+			{{ value }}
+		</component>
```

Comment:
This part is a bit subpar. `<input>` has no closing tag, normally. But … it’s a Vue template, not HTML. The rendered resulting HTML is valid again. 🤷

Reply:
That could be worth an inline comment. Paraphrasing or linking to your very comment here

Reply:
Good idea.